### ISSUE TITLE
rubycop.yml StyleDocumentation Enabled not working

### ISSUE THREAD

**Issue by ekampp** [rubocop-hq/rubocop#6238]
Hey

Using rubocop v `0.58.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin17)`

I'm getting a lot of hits from the documentation cop regarding top-level module and class documentation. 

I would expect this from my .rubocop.yml to disable that cop, but that made no difference.

```
Style/Documetation:
  Enabled: false
```

I have written this spec in documentation_spec.rb in my clone of the project, which also fails to pass:

```
context 'with globally disabled Style/Documentation' do
  let(:config) do
    RuboCop::Config.new('Style/Documentation' => {
                          'Enabled' => false
                        })
  end

  %w[class module].each do |keyword|
    it "accepts non-namespace #{keyword} without documentation" do
      expect_no_offenses(<<-RUBY.strip_indent)
        #{keyword} Test
          def method
          end
        end
      RUBY
    end
  end
end
```

I can see some old issues talking about this also, but they're [from 2013 and talks about the Style/Documentation being a top-level configuration](https://github.com/rubocop-hq/rubocop/issues/691) in the `.rubocop.yml`.

Thanks,

Emil

**Comment by Drenmi** [comment]
Hi, @ekampp.

```yml
Style/Documetation:
  Enabled: false
```

There's a typo here. `Documetation` is missing an "n". Please check whether this is the problem.

**Comment by ekampp** [comment]
Hi @Drenmi, 

Thanks for getting back to me. 

That was a typo in the bug report. my actual .rubocop.yml doesn't have the spelling error. As you can see, I can reproduce the problem with a spec as well, and that doesn't have the typo.

**Comment by jonas054** [comment]
@ekampp Disabling a cop in the cop specific spec file doesn't work, unfortunately. You have to put those kinds of tests in `cli_spec.rb`. You need something like this:
```ruby
  context 'with globally disabled Style/Documentation' do
    before do
      create_file('.rubocop.yml', <<-YAML.strip_indent)
        Style/Documentation:
          Enabled: false
      YAML
    end

    %w[class module].each do |keyword|
      it "accepts non-namespace #{keyword} without documentation" do
        create_file('file.rb', <<-RUBY.strip_indent)
          #{keyword} Test
            def method; end
          end
        RUBY
        expect(cli.run(%w[--format simple])).to eq(0)
        expect($stdout.string)
          .to eq(['', '1 file inspected, no offenses detected', ''].join("\n"))
      end
    end
  end
```
And that example passes for me. I wonder if you might inherit from some other configuration file where `Style/Documentation` is enabled. Run with the `--debug` flag to see which configuration files are loaded.
Do you get any warnings from `rubocop`, by the way?

**Comment by ekampp** [comment]
@jonas054 that's unexpected, but thanks for investigating. 

Which branch are you on?

**Comment by jonas054** [comment]
What I meant was that you can add an example like
```diff
diff --git a/spec/rubocop/cli_spec.rb b/spec/rubocop/cli_spec.rb
index dacda64..a9faa61 100644
--- a/spec/rubocop/cli_spec.rb
+++ b/spec/rubocop/cli_spec.rb
@@ -736,6 +736,28 @@ RSpec.describe RuboCop::CLI, :isolated_environment do
       RuboCop::ConfigLoader.default_configuration = nil
     end
 
+    context 'with globally disabled Style/Documentation' do
+      before do
+        create_file('.rubocop.yml', <<-YAML.strip_indent)
+          Style/Documentation:
+            Enabled: false
+        YAML
+      end
+
+      %w[class module].each do |keyword|
+        it "accepts non-namespace #{keyword} without documentation" do
+          create_file('file.rb', <<-RUBY.strip_indent)
+            #{keyword} Test
+              def method; end
+            end
+          RUBY
+          expect(cli.run(%w[--format simple])).to eq(0)
+          expect($stdout.string)
+            .to eq(['', '1 file inspected, no offenses detected', ''].join("\n"))
+        end
+      end
+    end
+
     context 'when a value in a hash is overridden with nil' do
       it 'acts as if the key/value pair was removed' do
         create_file('.rubocop.yml', <<-YAML.strip_indent)
```
on the latest master to demonstrate that disabling the `Style/Documentation` cop does indeed disable it. So there's nothing special with that cop. There must be something wrong with your configuration.
